src/hooks/index.test.ts

```diff
@@ -0,0 +1,254 @@
+import {expect} from 'chai'
+import * as sinon from 'ts-sinon'
+
+import {walletFactory} from '../../storybook/mocks/wallet'
+import {TxStatusRequest, TxStatusResponse} from '../types'
+import {fetchTxStatus} from '.'
+
+describe('fetchTxStatus', () => {
+  // it means that the tx was sent to node and processed
+  it('should return success when depth > 0', async () => {
+    // arrange
+    const txId = '12366dd454b4793d83d554329245d4bc67659f6f0468bb0e0d41f1ac6f496999'
+    const requestShouldBe: TxStatusRequest = {
+      txHashes: [txId],
+    }
+    const wallet = sinon.stubObject(walletFactory({}), {
```

Comment:
sinon is not necessary
stubbing/spying is not necessary
only the return value of fetchTxStatus matters

Reply:
I'm using stub so I can get the benefits of it, like the call count to ensure that it was retrying.